proto/baremetal/v1/host_pools_service.proto

```diff
@@ -0,0 +1,159 @@
+  // sort the host pools descending by API URL the value should be:
+  //
+  //     api_url desc
+  //
+  // If the parameter isn't provided, or if the value is empty, then the order of the results is undefined.
+  optional string order = 4;
+}
+
+message HostPoolsListResponse {
+  // Actual number of items returned. Note that this may be smaller than the value requested in the `limit` parameter
+  // of the request if there are not enough items, or of the system decides that returning that number of items isn't
+  // feasible or convenient for performance reasons.
+  optional int32 size = 1;
+
+  // Total number of items of the collection that match the search criteria, regardless of the number of results
+  // requested with the `limit` parameter.
+  optional int32 total = 2;
+
+  // List of results.
+  repeated HostPool items = 3;
+}
+
+message HostPoolsGetRequest {
+  string id = 1;
+}
+
+message HostPoolsGetResponse {
+  HostPool object = 1;
+}
+
+message HostPoolsCreateRequest {
+  HostPool object = 1;
+}
+
+message HostPoolsCreateResponse {
+  HostPool object = 1;
+}
+
+message HostPoolsUpdateRequest {
+  HostPool object = 1;
+  google.protobuf.FieldMask update_mask = 2;
+}
+
+message HostPoolsUpdateResponse {
+  HostPool object = 1;
+}
+
+message HostPoolsDeleteRequest {
+  string id = 1;
+}
+
+message HostPoolsDeleteResponse {}
+
+service HostPools {
+  // Retrieves the list of host pools.
+  rpc List(HostPoolsListRequest) returns (HostPoolsListResponse) {
+    option (google.api.http) = {get: "/api/baremetal/v1/host_pools"};
+  }
+
+  // Retrieves the details of one specific host pool.
+  rpc Get(HostPoolsGetRequest) returns (HostPoolsGetResponse) {
+    option (google.api.http) = {
+      get: "/api/baremetal/v1/host_pools/{id}"
+      response_body: "object"
+    };
+  }
+
+  // Creates a new host pool.
+  //
+  // Note that this operation is not allowed for regular users, only for the server. Regular users create host pools
+  // indirectly, creating a host pool order that will eventually result in the system creating a host pool.
+  rpc Create(HostPoolsCreateRequest) returns (HostPoolsCreateResponse) {
+    option (google.api.http) = {
+      post: "/api/baremetal/v1/host_pools"
+      body: "object"
+      response_body: "object"
+    };
+  }
+
+  // Updates an existing host pool.
+  //
+  // In the HTTP+JSON version of the API this is mapped to the `PATCH` verb and the `update_mask` field is automatically
+  // populated from the list of fields present in the request body. For example, to update the `state` of a host pool to
+  // `READY` the request line should be like this:
+  //
+  // ```http
+  // PATCH /api/baremetal/v1/host_pools/123
+  // ```
+  //
+  // And the request body should be like this:
+  //
+  // ```json
+  // {
+  //   "status": {
+  //     "state": "HOST_POOL_STATE_READY"
+  //   }
+  // }
+  // ```
```

Comment:
This example isn't ideal because users won't be able to update the state of a host pool, I think. It would be better to use an example of how to update the size of the node set of the pool.

Reply:
Done!